src/System.Private.CoreLib/shared/System/MemoryExtensions.cs

```diff
@@ -1653,14 +1654,17 @@ public static bool Overlaps<T>(this ReadOnlySpan<T> span, ReadOnlySpan<T> other,
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static int BinarySearch<T, TComparer>(
             this ReadOnlySpan<T> span, T value, TComparer comparer)
-            where TComparer : IComparer<T>
+            where TComparer : IComparer<T>?
         {
+
             if (comparer == null)
-                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.comparer);
```

Comment:
If we are going to do this, the xml comments need to be updated (since we no longer throw for a `null` comparer).

Reply:
Good catch.